### PR TITLE
Fixing the issue of getting "Unknown" value with platform.processor

### PR DESCRIPTION
**Pull request by Naresh-ibm**
platform.processor is returning the unknown value and hence tests were failing. So changed this checking with other util functions like cpu.get_arch()

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
```
Before fix
-------------

# avocado run --max-parallel-tasks=1 pci_hotplug.py -m pci_hotplug.py.data/pci_hotplug.yaml 
JOB ID     : 34e727d98d44339e59dcdc2dbc0c229c84e96d1b
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-09-19T04.47-34e727d/job.log
 (1/1) pci_hotplug.py:PCIHotPlugTest.test;run-peer_ip-3dcc: STARTED
 (1/1) pci_hotplug.py:PCIHotPlugTest.test;run-peer_ip-3dcc: CANCEL: Processor is not ppc64 (0.01 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-09-19T04.47-34e727d/results.html
JOB TIME   : 23.01 s

After fix
-------------

]# avocado run --max-parallel-tasks=1 pci_hotplug.py -m pci_hotplug.py.data/pci_hotplug.yaml
JOB ID     : c857674a96c01dc50eb6dffd3037496c1648d361
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-09-19T05.47-c857674/job.log
 (1/1) pci_hotplug.py:PCIHotPlugTest.test;run-peer_ip-3dcc: STARTED
 (1/1) pci_hotplug.py:PCIHotPlugTest.test;run-peer_ip-3dcc: PASS (1.28 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-09-19T05.47-c857674/results.html
JOB TIME   : 115.84 s


```

**Comment by abdhaleegit** [comment]
@Naresh-ibm Please wait for https://github.com/avocado-framework/avocado/pull/5993 when this is merged.. you can close this PR.. as this is same fix

File: io/pci/pci_hotplug.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def setUp(self):
         """
         Setup the device.
         """
-        if 'ppc' not in platform.processor():
+        if 'ppc' not in cpu.get_arch():
```
Comment (abdhaleegit) [inline io/pci/pci_hotplug.py:49]: cpu.get_arch will actually returns `powerpc` and not ppc 

https://github.com/avocado-framework/avocado/blob/3b1f5cbbb07ea850498b55aa020d74cc0f7fa15f/avocado/utils/cpu.py#L200C31-L200C38

File: memory/memhotplug.py [hunk 2]
```diff
@@ -195,7 +195,7 @@ def test_hotplug_toggle(self):
         self.__error_check()
 
     def test_dlpar_mem_hotplug(self):
-        if 'ppc' in platform.processor() and 'PowerNV' not in open('/proc/cpuinfo', 'r').read():
+        if 'ppc' in cpu.get_arch() and 'PowerNV' not in open('/proc/cpuinfo', 'r').read():
```
Comment (abdhaleegit) [inline memory/memhotplug.py:198]: I see the platform apis returning value correctly

```
>>> import platform
>>> platform.processor()
'ppc64le'
```

which distro has this problem ?

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

File: io/pci/pci_hotplug.py (hunk 1 above)
Comment (Naresh-ibm) [reply io/pci/pci_hotplug.py:49]:
```
>>> from avocado.utils import cpu
>>> cpu.get_arch()
'ppc64le'
>>> 
```

File: memory/memhotplug.py (hunk 2 above)
Comment (abdhaleegit) [reply memory/memhotplug.py:198]: @Naresh-ibm is this related https://github.com/avocado-framework/avocado/pull/5993

Comment (Naresh-ibm) [reply memory/memhotplug.py:198]: This is python issue sir.
And I think the above util PR is different.

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Pull request merged**